crates/core/src/operations/transaction/protocol.rs

```diff
@@ -61,11 +61,11 @@ impl ProtocolChecker {
     }
 
     pub fn default_reader_version(&self) -> i32 {
-        1
+        3
     }
 
     pub fn default_writer_version(&self) -> i32 {
-        2
+        7
```

Comment:
This is a non-trivial change in behavior isn't it? I don't think timezone support gets us 3/7

Reply:
Correct, but it smoothes out our table creation versioning since feature toggling in 3,7 is more flexible

Reply:
Hmm, what's your worry? 

I don't see the issue of automatically enabling a feature during create based on the source data. It's something we can easily infer, if it contains a timezone, then enable the feature and bump the protocol versions.

What I can do, is undo the change of the default protocol version back to 1,2. However I still believe it's more convenient if it's by default on 3,7 especially since we started supporting 3,7 so that we could pick and choose on which table features we added support instead of going linear from 2 to 7.

Reply:
because doing this breaks all readers/writers in the wild that aren't 3/7, it breaks prior versions of delta-spark as well, even if you never wrote a single row to the table because it requires readers and writers to meet that bar to even read the table. I'm saying, we should only update the protocol if the features are explicitly enabled or explicitly used. For example, databricks enables deletion vectors by default which makes basically the default protocol 3,7, but if you disable that in the table create it sets the protocol to 1,2. The protocol versions should be the absolute minimum required to read/write the table, we should never just update them if there isn't a feature that requires it.

Reply:
Gotcha, I see your point now, that will definitely break backwards compatibility indeed with older databricks runtimes. 

Then I'll change back the default to 1,2 but bump it only  during a protocol commit when we see a feature that is only on 3,7.

Reply:
Excellent, so when you create a table with timestamp_ntz, it will create a table with 3,7 as the versions and timestampNtz in the table_features

Reply:
I have undone the changes :) So it's now only done while writing a table with no timezone.